drivers/zfs/zfs.go

```diff
@@ -109,9 +111,6 @@ func Init(base string, opt []string, uidMaps, gidMaps []idtools.IDMap) (graphdri
 		return nil, fmt.Errorf("Failed to create '%s': %v", base, err)
 	}
 
-	if err := mount.MakePrivate(base); err != nil {
-		return nil, err
-	}
```

Comment:
Are we sure that this is something that we no longer need to do for zfs filesystems for our use cases?

Reply:
I don't know.

Reply:
Added it back, since other drivers all do it.  Will remove if the problem comes back to the user.

Reply:
Here's the Moby commit removing the code: https://github.com/moby/moby/commit/9803272f2db84df7955b16c0d847ad72cdc494d1

Reply:
Well that is the exact code I added back, and that is the difference between the patch he tested and the final result.

The reason that code was in the original storage drivers is to prevent leaked mount points from propagating out of the storage directory.  @rhvgoyal I believe questions if we still need it.  I say remove it to fix this issue and then see if we see mount leak problems in the future.